lib/logstash/outputs/logstash.rb

```diff
@@ -10,8 +10,18 @@ class LogStash::Outputs::Logstash < LogStash::Outputs::Base
 
   config_name "logstash"
 
-  config :host,     :validate => :string,   :required => true
-  config :port,     :validate => :number,   :required => true
+  # Sets the host of the downstream Logstash instance.
+  # Host can be any of the forms:
+  #     `"127.0.0.1"`
+  #     `"127.0.0.1:9800"`
+  #     `"http://127.0.0.1"`
+  #     `"https://127.0.0.1:9800"`
+  # If the protocol is unspecified, this plugin assumes `https` when `ssl_enabled => true` (default)
+  # or `http` when `ssl_enabled => false`.
+  #
+  # NOTE: `hosts` naming is intentional and multi-host support is planned.
+  #
+  config :hosts, :validate => :uri, :required => true
```

Comment:
I don't think we can use `:validate => :uri` here, because it exposes internal implementation details (transport protocol).

We could add a `host_port_pair` validator to the `ValidatorSupport` mixin (pseudo-code based on its [`FieldReferenceValidationAdapter`](https://github.com/logstash-plugins/logstash-mixin-validator_support/blob/2cf8b5ce398e6b9f0de46f2e33921e8d9bb64511/lib/logstash/plugin_mixins/validator_support/field_reference_validation_adapter.rb)):

~~~ ruby
# encoding: utf-8

require 'logstash/plugin_mixins/validator_support'

module LogStash
  module PluginMixins
    module ValidatorSupport

      host_port_regex = %r{\A
        (?:<host>(?:(?:[a-z][a-z0-9\-\.]*) # lax domain name
                   |(?:[0-9\.])            # lax dot-decimal ipv4
                   |(?:\[[0-9a-f:\.]+\])   # lax bracket-enclosed ipv6
                   )
        )
        :
        (?:<port>[0-9]{1,5}) # lax port number
      \z}xi

      
      HostPortPairValidationAdapter = NamedValidationAdapter.new(:host_port_pair) do |value|
        break ValidationResult.failure("Expected exactly one host-port pair, got `#{value.inspect}`") unless value.kind_of?(Array) && value.size <= 1

        candidate = value.first

        break ValidationResult.failure("Expected a valid host:port pair, got `#{candidate.inspect}`") unless host_port_regex =~ candidate

        break ValidationResult.success(candidate)
      end
    end
  end
end
~~~

Which we could then include into this plugin with something like:

~~~ ruby
require 'logstash/plugin_mixins/validator_support/host_port_pair_validation_adapter'

class LogStash::Inputs::Foo < Logstash::Inputs::Base
  extend LogStash::PluginMixins::ValidatorSupport::HostPortPairValidationAdapter

  config :hosts, :validate => :host_port_pair, :list => true, : required => true
end
~~~

(Note: validators get called once for each _value_ so a config specified as `:list => true` will run the validator for each entry independently. We may also need to validate that the list itself is not empty)

Reply:
Thanks @yaauie for the suggestion. I have updated (optional port, stricter regex and unit tests) this pseudo and added with following PR: https://github.com/logstash-plugins/logstash-mixin-validator_support/pull/4
Your review is always appreciated!